mocks/arbeidsforhold-mock/src/main/java/no/nav/tjeneste/virksomhet/arbeidsforhold/v3/ArbeidsforholdMockImpl.java

```diff
@@ -115,7 +119,11 @@ public FinnArbeidsforholdPrArbeidstakerResponse finnArbeidsforholdPrArbeidstaker
                 List<no.nav.tjeneste.virksomhet.arbeidsforhold.v3.informasjon.arbeidsforhold.Arbeidsforhold> responseArbeidsforhold = new ArrayList<>();
 
                 for (Arbeidsforhold arbeidsforhold : arbeidsforholdModell.getArbeidsforhold()) {
-                    responseArbeidsforhold.add(arbeidsforholdAdapter.fra(fnr, arbeidsforhold));
+                    if (request.getArbeidsforholdIPeriode() != null && erOverlapp(request.getArbeidsforholdIPeriode(), arbeidsforhold)) {
+                        responseArbeidsforhold.add(arbeidsforholdAdapter.fra(fnr, arbeidsforhold));
+                    } else if (request.getArbeidsforholdIPeriode() == null) {
+                        responseArbeidsforhold.add(arbeidsforholdAdapter.fra(fnr, arbeidsforhold));
+                    }
```

Comment:
Hvorfor er det to if statements som gjør det samme?

Reply:
Den eine sjekker på om perioden er satt og overlapper medan den andre sjekker på om den ikkje er satt.

Reply:
Reagerte bare på at ved begge utfallene så vil "responseArbeidsforhold.add(arbeidsforholdAdapter.fra(fnr, arbeidsforhold));" bli kjørt. Så hvorfor ha 2 if statement? Er det fordi det er ryddigere?

Reply:
Har endret til å sjekke alt i if-statement